internal/gateways/identity_api.go

```diff
@@ -0,0 +1,104 @@
+package gateways
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/DIMO-Network/edge-network/internal/models"
+	"github.com/DIMO-Network/shared"
+	"github.com/pkg/errors"
+	"io"
+	"time"
+)
+
+//go:generate mockgen -source identity_api.go -destination mocks/identity_api_mock.go
+type IdentityApi interface {
+	QueryIdentityAPIForVehicles(ethAddress string) ([]models.VehicleInfo, error)
+}
+
+type identityAPIService struct {
+	httpClient shared.HTTPClientWrapper
+}
+
+const IdentityAPIURL = "https://identity-api.dimo.zone/query"
+
+func NewIdentityAPIService() IdentityApi {
+	h := map[string]string{}
+	h["Content-Type"] = "application/json"
+	hcw, _ := shared.NewHTTPClientWrapper("", "", 10*time.Second, h, false) // ok to ignore err since only used for tor check
+
+	return &identityAPIService{
+		httpClient: hcw,
+	}
+}
+
+func (i *identityAPIService) QueryIdentityAPIForVehicles(ethAddress string) ([]models.VehicleInfo, error) {
+	// GraphQL query
+	graphqlQuery := `{
+        vehicles(first: 10, filterBy: { owner: "` + ethAddress + `" }) {
```

Comment:
@JamesReate , suggest what value for the first we should use. I put 10, but per my understanding, we always should have only one element in the array back. If the owner has multiple vehicles/devices, how can we filter the devices we care about?

Reply:
ah wait, my understanding is this ethAddress (one we get from device) is the device eth addr. so the query would be based from the aftermarketDevice. 
ie. query would be like this - and that token is the vehicleTokenID (maybe good to make the naming specific for return value):
```
{
  aftermarketDevice(by: {address: "0x"}) {
    vehicle {
      tokenId
    }
  }
}
```

Reply:
```
{
  aftermarketDevice(by: {address: "0x"}) {
    vehicle {
      tokenId,
      definition {
        make
        model
        year
      }
    }
  }
}
```
like this with the make model year

Reply:
Gotcha, so the response from aftermarketDevice will always be a single object, not an array. I need to get more familiar with graphql.